[codex] Fix Feature Flags documentation guidance (#36573)
DataDog/documentation: content/en/feature_flags/_index.md

## Overview

> Datadog Feature Flags is Datadog's flag management product. You create flags in Datadog, deliver flag configuration to Datadog SDKs, and evaluate variants in your application through Datadog or OpenFeature APIs.

Author: 🤖 **Context from Codex:** This establishes the distinction the old hub was missing: Datadog Feature Flags is the flag-management product, while RUM Feature Flag Tracking is the observability path for third-party providers. Without that distinction, browser telemetry settings later in the docs are easy to misread.

## Overview

Datadog Feature Flags is Datadog's flag management product. You create flags in Datadog, deliver flag configuration to Datadog Feature Flags SDKs, and evaluate variants in your application through Datadog or OpenFeature APIs.

> Datadog Feature Flags is built on the [OpenFeature standard](https://openfeature.dev/docs/reference/intro/), an open-source, vendor-neutral specification for feature flag APIs. See OpenFeature's getting-started guide if you're new to OpenFeature concepts like providers, evaluation context, and hooks.

Author: 🤖 **Context from Codex:** Adds the OpenFeature primer link on the hub so readers can learn provider, evaluation-context, and hook concepts from the upstream standard instead of us duplicating that primer here. Source: https://openfeature.dev/docs/reference/intro/.

> Datadog Feature Flags is built on the [OpenFeature standard](https://openfeature.dev/docs/reference/intro/), an open-source, vendor-neutral specification for feature flag APIs. See OpenFeature's getting-started guide if you're new to OpenFeature concepts like providers, evaluation context, and hooks.

Reviewer: "OpenFeature's getting-started guide" is presented as a navigation hint but is not linked. If there are specific OpenFeature docs pages that users should read, we should link to them directly; otherwise they could get frustrated having to navigate an entirely separate set of docs. I'd either add a link or remove the sentence (it's somewhat redundant with the preceding link).

(This paragraph appears verbatim on three pages — `feature_flags/_index.md:32`, `client/_index.md:22`, `server/_index.md:17`. Apply the fix to all three.)

Author: 🤖 **Response from AI:** Applied to all three hub pages (`feature_flags/_index.md`, `client/_index.md`, `server/_index.md`). Rather than drop the sentence, the concepts reference now links to the [OpenFeature concepts documentation](https://openfeature.dev/docs/category/concepts), which is a distinct page from the intro already linked in the first sentence (so no duplicate-link redundancy).

Feature flags enable you to toggle features on and off, conduct A/B/n testing, gradually roll out new functionality, and personalize user experiences without the need for extensive code deployments. With feature flags, you can empower your team to make dynamic changes, iterate rapidly, and deliver enhanced user experiences.

If your flags are managed by LaunchDarkly, Split, ConfigCat, or another provider and you only want Datadog to record evaluated variants in RUM, see [RUM Feature Flag Tracking](/real_user_monitoring/feature_flag_tracking/) instead.

> Use a client-side SDK when the flag must be evaluated in a browser, mobile app, or game client. Use a server-side SDK when the decision should happen in a backend service and the flag configuration can be delivered through the Datadog Agent and Remote Configuration.

Reviewer: More explicit language will help readability for both AI and humans:

```suggestion
Use a client-side SDK when the flag is evaluated in a browser, mobile app, or game client. Use a server-side SDK when the flag is evaluated in a backend service that receives flag configuration through the Datadog Agent and Remote Configuration.
```

### Credentials at a glance

| Credential | Used by | Where it goes | Sensitive? |
| --- | --- | --- | --- |

> | Client token | Browser, mobile, and game SDKs | Client application configuration | Public-shipping token |

Reviewer: "Public-shipping token"/"Public-shipping identifier" is non-standard Datadog terminology and the compound adjective is awkward. Suggest clearer values for the "Sensitive?" column:

```suggestion
| Client token | Browser, mobile, and game SDKs | Client application configuration | No — safe to ship in public client code |
| Application ID | Browser and RUM-backed client SDKs | Client application configuration | No — public identifier |
| API key | Datadog Agent for server-side Remote Configuration | Agent configuration only | Yes — keep server-side only |
```

The same table also appears at `content/en/getting_started/feature_flags/_index.md:216–217` and would benefit from the same edit.

Author: 🤖 **Response from AI:** Applied. The batch commit added the clearer rows but left the old `Public-shipping identifier` / `Secret` rows in place, so the table had duplicates. Removed the leftover rows; the table now has exactly three rows with the suggested `No — safe to ship in public client code` / `No — public identifier` / `Yes — keep server-side only` values.

| Application ID | Browser and RUM-backed client SDKs | Client application configuration | Public-shipping identifier |
| API key | Datadog Agent for server-side Remote Configuration | Agent configuration only | Secret |

Do not put API keys in browser, mobile, or game applications.

### Evaluation context and telemetry

Evaluation context attributes are the flat string, number, and Boolean values that Datadog uses for targeting rules and rollout bucketing. Set a stable `targetingKey`, such as a user ID, session ID, or device ID, so percentage rollouts are consistent.

> Feature Flags telemetry includes exposure events, flag evaluation metrics, and optional RUM correlation depending on the SDK and configuration.

Author: 🤖 **Context from Codex:** Removes the second RUM Feature Flag Tracking pointer from the telemetry section because the index already directs third-party flag-provider users to that page earlier in the overview.
